### PR TITLE
fix(#633): handle invalid UUIDs in found_by parameter

### PR DESCRIPTION
**Pull request by stefopl**

### CODE REVIEW

File: okapi/services/caches/search/SearchAssistant.php [hunk 1]
```diff
@@ -423,10 +423,29 @@ public function prepare_common_search_params()
             try {
                 $users = OkapiServiceRunner::call("services/users/users", new OkapiInternalRequest(
                     $this->request->consumer, null, array('user_uuids' => $tmp, 'fields' => 'internal_id')));
+	            if (empty($users) || !is_array($users)) {
```
Comment (wrygiel) [inline okapi/services/caches/search/SearchAssistant.php:426]: This should be outside (below) the surrounding try..catch.

File: okapi/services/caches/search/SearchAssistant.php [hunk 2]
```diff
@@ -423,10 +423,29 @@ public function prepare_common_search_params()
...
+		        }
+	        }
+
+	        if (!empty($invalid_uuids)) {
+		        throw new InvalidParam(
+			        'found_by',
+			        "The following UUID(s) are invalid or not found: " . implode(", ", $invalid_uuids)
+		        );
+	        }
+
```
Comment (wrygiel) [inline okapi/services/caches/search/SearchAssistant.php:448]: Indent with spaces (match the existing indentation).

**Review by wrygiel — APPROVED** [review]
Thanks for the fix! This looks good to me (apart from some details mentioned in the review). However, I no longer actively maintain OKAPI, so I suggest you wait at least one day before merging (perhaps there are other, more active reviewers who'd want to chip in).

**Comment by stefopl** [comment]
It would be best if the fix waited for merging at least until Monday.
And if someone without a development environment would like to check, they can go to ocpl .stefo .pl/ okapi/

**Pull request merged**